netcdf.yaml

```diff
@@ -0,0 +1,80 @@
+package:
+  name: netcdf
+  version: 4.9.2
+  epoch: 0
+  description: Libraries for the Unidata network Common Data Format
+  copyright:
+    - license: BSD-3-Clause
+
+environment:
+  contents:
+    packages:
+      - bash
+      - build-base
+      - busybox
+      - bzip2-dev
+      - cmake
+      - curl-dev
+      - hdf5-dev
+      - libaec-dev
+      - libxml2-dev
+      - m4
+      - samurai
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/Unidata/netcdf-c
+      tag: v${{package.version}}
+      expected-commit: 9328ba17cb53f13a63707547c94f4715243dafdf
+
+  - uses: patch
+    with:
+      patches: gcc14.patch
+
+  - runs: |
```

Comment:
can we consider using the pipeline here? something like following:
```yaml
  - uses: cmake/configure
    with:
      opts: |
        -DCMAKE_BUILD_TYPE=None \
        -DENABLE_PERL=OFF \
        -DENABLE_TESTS=OFF \
        -DWITH_PCRE2=True
```

afterwards, we can use `cmake/build` and `cmake/install` command. 
You can refer to openscap melange config (ref:https://github.com/wolfi-dev/os/blob/main/openscap.yaml) for reference.

Reply:
@kranurag7 Done 👍